src/Model/Table/PreciousUsersTable.php

```diff
@@ -12,10 +15,34 @@ public function initialize(array $config)
 
     /**
      * 大切な人一覧を取得するクエリを取得
-     * @return \Cake\ORM\Query
+     * @return Query
      */
     public function findPreciousUsers()
     {
         return $this->find();
     }
+
+    /**
+     * 大切な人を登録する
+     * @param array $precious_user
+     * @return Entity|boolean
+     */
+    public function addPreciousUser(array $precious_user)
```

Comment:
引数arrayにしちゃうとarrayの中の型が分からなくなる&必要な値が入ってないなどが起きうるので
newEntityはController側でやってしまって、ここでは `PreciousPeole` エンティティを渡してあげるのがいいかなと思ったんですが、どうでしょうか？

それか、引数を明示的にしてあげるとかが良さそうです！
```suggestion
    public function addPreciousUser(string $name, int $user_id, ...)
```

Reply:
それか、
`src/Controller/Api/V1/FavoriteProductsController.php`
あたりを参考にしつつ、Formをうまく使ってやるとかはありそうですね

Reply:
元々は、controllerから直接tableのメソッドを呼ばない構成にしたいという意図でarrayにしていたのですが、林くんの指摘通りarrayだとデメリットの方が大きいですね、、、

ということでcontrollerでもtableのnewEntityを使うのが良さそうだと思いました！
（cakephpではEntity = ドメインオブジェクト的な役割みたいですね）

Reply:
Formを使うやり方も参考になりました！

Reply:
ありがとうございます！

> 元々は、controllerから直接tableのメソッドを呼ばない構成にしたいという意図でarrayにしていた

本当はこの方がいいですね！なのでFormを使ったりうまくやるのが良さそうですね！
ありがとうございます！